tempodb/backend/s3/s3.go

```diff
@@ -637,7 +636,7 @@ func (rw *readerWriter) readAll(ctx context.Context, name string) ([]byte, error
 func (rw *readerWriter) readAllWithObjInfo(ctx context.Context, name string) ([]byte, minio.ObjectInfo, error) {
 	options := getObjectOptions(rw)
 	reader, info, _, err := rw.hedgedCore.GetObject(ctx, rw.cfg.Bucket, name, options)
-	if err != nil && minio.ToErrorResponse(err).Code == s3.ErrCodeNoSuchKey {
+	if err != nil && minio.ToErrorResponse(err).Code == "NoSuchKey" {
```

Comment:
```suggestion
	if err != nil && minio.ToErrorResponse(err).Code == minio.NoSuchKey {
```

We can use the constant defined un minio